src/stcal/jump/jump.py

```diff
@@ -262,3 +299,131 @@ def detect_jumps(frames_per_group, data, gdq, pdq, err,
+                        enclosing ellipse or circle
+    :param expand_factor: The factor that increases the size of the snowball or enclosing ellipse.
+    :param use_ellipses:  Use ellipses rather than circles (better for MIRI)
+    :param sat_required_snowball: Require that there is a saturated pixel within the radius of the jump
+                        circle to trigger the formation of a snowball.
+    :return: none
+    """
+    for integration in range(gdq.shape[0]):
+        for group in range(1, gdq.shape[1]):
+            if use_ellipses:
+                jump_ellipses = find_ellipses(gdq[integration, group, :, :], jump_flag, min_jump_area)
+                log.info(f"Integration {integration} group {group} number of showers = {len(jump_ellipses)}.")
+                gdq[integration, group, :, :], num_events = \
+                    extend_ellipses(gdq[integration, group, :, :],
+                                    jump_ellipses, sat_flag, jump_flag,
+                                    expansion=expand_factor)
+            else:
+                new_flagged_pixels = gdq[integration, group, :, :] - gdq[integration, group - 1, :, :]
+                sat_circles = find_circles(new_flagged_pixels, sat_flag, min_sat_area)
+                sat_pixels = np.bitwise_and(new_flagged_pixels, sat_flag)
+                saty, satx = np.where(sat_pixels == sat_flag)
+                only_jump = gdq[integration, group, :, :].copy()
+                only_jump[saty, satx] = jump_flag
+                jump_circles = find_circles(only_jump, jump_flag, min_jump_area)
+                if sat_required_snowball:
+                    snowballs = make_snowballs(jump_circles, sat_circles)
+                else:
+                    snowballs = jump_circles
+                log.info(f"Integration {integration} group {group} number of snowballs = {len(snowballs)}.")
+                gdq[integration, group, :, :], num_events = extend_snowballs(gdq[integration, group, :, :],
+                                                                             snowballs, sat_flag,
+                                                                             jump_flag,
+                                                                             expansion=expand_factor)
+
+
+def extend_snowballs(plane, snowballs, sat_flag, jump_flag, expansion=1.5):
+    # For a given DQ plane it will use the list of snowballs to create expanded circles of pixels with
+    # the jump flag set.
+    image = np.zeros(shape=(plane.shape[0], plane.shape[1], 3), dtype=np.uint8)
+    num_circles = len(snowballs)
+    sat_pix = np.bitwise_and(plane, 2)
+    for snowball in snowballs:
+        jump_radius = snowball[1]
+        jump_center = snowball[0]
+        cenx = jump_center[1]
+        ceny = jump_center[0]
+        extend_radius = round(jump_radius * expansion)
+        image = cv.circle(image, (round(ceny), round(cenx)), extend_radius, (0, 0, 4), -1)
```

Comment:
why are the coordinates for the center of the circle being rounded?

Reply:
Because cv.circle requires them to be integers. I think the work is done by C routines which need specific types.